File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_1 = {}
        ans=[]
        i=0
        while(i<len(nums)):
            v = target-nums[i]
            value = dict_1.get(v,"-1")
            if value!= "-1":
                ans.append(value)
                ans.append(i)
                break
            else:
                dict_1[nums[i]]=i
            i+=1
        return ans

        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer,Integer> nmap = new HashMap<>();
        int n = nums.length;

        for(int i =0;i<n;i++){

            nmap.put(nums[i],i);
        }

        for(int i=0;i<n;i++){

            int complement = target -nums[i];
            if(nmap.containsKey(complement)  && nmap.get(complement)!=i)
                   return new int[]{i,nmap.get(complement)};
        }
                return new int[]{}; // No solution found

        
    }
}